averbis/core/_rest_client.py

```diff
@@ -2566,3 +2709,58 @@ def _create_zip_io(source: Union[IO, Path, str], path_in_zip: Union[Path, str] =
 
         zip_archive.seek(0)
         return zip_archive
+
+    @staticmethod
+    def _fetch_text_analysis_result_filename(source: Union[Path, IO, str]) -> str:
+        if isinstance(source, Path):
+            return Path(source).name
+
+        if isinstance(source, IOBase) and hasattr(source, "name"):
+            return source.name
+
+        if isinstance(source, Cas):
+            return "annotated_document.xmi"
+
+        raise ValueError("Unsupported source type - valid is [Path, IO, CAS]")
+
+    def _add_text_analysis_result_to_document(self, project_name, process_name, document_collection_name, source,
+                                              document_name, mime_type, typesystem):
+        filename = self._fetch_text_analysis_result_filename(source)
+        files, params = self._create_text_analysis_request_parts(document_name, filename, source, mime_type, typesystem)
+
+        self.__request_with_json_response(
+            "post",
+            f"/experimental/textanalysis/projects/{project_name}/documentCollections/{document_collection_name}/processes/{process_name}/textAnalysisResult",
+            files=files, params=params
+        )
+
+    @staticmethod
+    def _create_text_analysis_request_parts(document_name, filename, source, mime_type, typesystem):
+        if isinstance(source, Cas):
```

Comment:
I believe we have similar code also in `_import_documents` - can we refactor both places so they use the same code?